wap/loaders/productListingPage.ts

```diff
@@ -0,0 +1,140 @@
+import { ProductListingPage } from "../../commerce/types.ts";
+import { AppContext } from "../mod.ts";
+import {
+  getUrl,
+  toBreadcrumbList,
+  toFilters,
+  toProduct,
+} from "../utils/transform.ts";
+import { WapProductsListPage } from "../utils/type.ts";
+import { TypedResponse } from "../../utils/http.ts";
+
+export interface Props {
+  busca?: string;
+  /**
+   * @max 100
+   * @default 12
+   */
+  limit?: number;
+}
+
+const endPoint = {
+  "product/listing/category":
+    "GET /api/v2/front/url/product/listing/category" as const,
+  "product/listing/brand":
+    "GET /api/v2/front/url/product/listing/brand" as const,
+  "product/listing/landing-page":
+    "GET /api/v2/front/url/product/listing/landing-page" as const,
+};
+
+/**
+ * @title Wap Integration
+ * @description Product List loader
+ */
+const loader = async (
+  props: Props,
+  req: Request,
+  ctx: AppContext,
+): Promise<ProductListingPage | null> => {
+  const { api } = ctx;
+  const { url: baseUrl } = req;
+  const url = new URL(baseUrl);
+
+  const rawSearch = url.searchParams.get("busca") ?? props.busca;
+  const busca = rawSearch && encodeURIComponent(rawSearch);
+
+  const page = Number(url.searchParams.get("pg") || 1);
+
+  const limit = Number(url.searchParams.get("ipp") ?? props.limit ?? 12);
+
+  const offset = page <= 1 ? 0 : (page - 1) * limit;
+
+  if (!busca && url.pathname.startsWith("%2F_fresh")) return null;
+
+  const { nivel } = await api
+    ["GET /api/v2/front/url/verify"]({
+      url: url.pathname,
+    }).then((response) => response.json()).catch(() => {}) ?? {};
+
+  if (!busca && !nivel) return null;
+
+  const params: Record<string, string | string[]> = {};
+  url.searchParams.delete("busca");
+  url.searchParams.delete("pg");
+  url.searchParams.forEach((v, k) => {
```

Comment:
yes, the problem here its: I could'nt find a pattern betewen every filter, so map all they its realy hard, but I can add the basic filters as panel props, the page prop i just dont add because i dont see a real use case

Reply:
Nice, page prop is nott necessary I think. The basic filters should be good, you can open a issue.